packages/@aws-cdk/cloudformation-include/test/test-templates/functions-and-conditions.json

```diff
@@ -0,0 +1,122 @@
+{
+    "Mappings" : {
+      "RegionMap" : {
+        "region-1" : { 
+          "HVM64" : "name1", "HVMG2" : "name2" 
+        }
+      }
+    },
+    "Conditions": {
+        "AlwaysTrueCond": {
+          "Fn::Not": [{
+            "Fn::Equals": [{"Ref": "AWS::Region"}, "completely-made-up-region"]}]
+        },
+        "AlwaysFalseCond": {
+          "Fn::Equals": [{ "Ref": "AWS::Region" }, "completely-made-up-region"]
+        },
+        "AndCond": {
+          "Fn::And": [
+            {"Condition": "AlwaysTrueCond"},
+            {"Condition": "AlwaysFalseCond"}
+          ] 
+        },
+        "OrCond": {
+          "Fn::Or": [
+            {"Condition": "AlwaysTrueCond"},
+            {"Condition": "AlwaysFalseCond"}
+          ] 
+        }
+      },
+    "Resources": {
+      "Vpc": {
+        "Type" : "AWS::EC2::VPC",
+        "Properties" : {
+          "CidrBlock" : {
+              "Fn::If": [
+                  "AlwaysTrueCond",
+                  { "Fn::Cidr": [ "192.168.1.1/24", 2] },
+                  { "Fn::Cidr": [ "10.0.0.0/24", 6, 5] }
+              ]
+          }
+        }
+      },
+      "Bucket": {
+        "Type": "AWS::S3::Bucket",
+        "Properties": {
+          "BucketName": {
+            "Fn::If": [
+              {"Fn::Or": ["AlwaysFalseCond", "AlwaysFalseCond"]},
+              "Unreachable",
+              {"Fn::FindInMap": [
+                "RegionMap",
+                "region-1",
+                "HVM64"
+              ]}
+            ]
+          }
+        }
+      },
+      "Subnet1" : {
+        "Type" : "AWS::EC2::Subnet",
+        "Properties" : {
+          "VpcId" : { 
+            "Fn::If": [
+              {"Fn::And": ["AlwaysTrueCond", "AlwaysTrueCond"]},
+              { 
+                "Fn::Split": [
+                  ",",
+                  { "Fn::ImportValue" : "ImportedVpcId" }
+              ]},
+              "Unreachable"
+            ]
+          },
+          "CidrBlock" : "10.0.0.0/24",
+          "AvailabilityZone" : {
+            "Fn::Select" : [ 
+              "0", 
+              { 
+                "Fn::GetAZs" : "" 
+              } 
+            ]
+          }
+        }
+      },
+      "Subnet2" : {
+        "Type" : "AWS::EC2::Subnet",
+        "Properties" : {
+          "VpcId" : { 
+            "Fn::Select" : ["3", ["foo", "aValidVPCID", "foo2"]]
+          },
+          "CidrBlock" : "10.0.0.0/24",
+          "AvailabilityZone" : {
+            "Fn::Select" : [ 
+              "0", 
+              { 
+                "Fn::GetAZs" : "eu-west-2" 
+              } 
+            ]
+          }
+        }
+      },
+      "TransformBucket": {
+        "Type": "AWS::S3::Bucket",
+        "Properties": {
+          "BucketName": {
+            "Fn::If": [
+              { "Fn::Or": ["AlwaysFalseCond", "AlwaysTrueCond"]}, {
```

Comment:
This isn't valid CloudFormation (`Fn::If` takes a condition name as its first argument).